src/features/messages/cards/TransactionCard.tsx

```diff
@@ -236,11 +249,59 @@ function CallDataModal({ debugResult }: { debugResult?: MessageDebugResult }) {
           </p>
           <LabelAndCodeBlock label="Recipient contract address:" value={contract} />
           <LabelAndCodeBlock label="Handle function input calldata:" value={handleCalldata} />
+          <button onClick={handleClick}
+            disabled={loading}
+            className='underline text-blue-400'
+          >
+            {buttonText}
+          </button>
+          {loading && <Spinner classes="mt-4 scale-75 self-center" />}
         </div>
       </Modal>
     </>
   );
 }
+async function simulateCall({contract,handleCalldata,chainId,message}:{contract:string,handleCalldata:string,chainId:ChainId,message:Message}){
+
```

Comment:
Wrap the function contents here in a `try/catch` and show an error message with `toast.error` on failure

Reply:
Didn't wrap this in `try/catch`, rather wrapped the backend call which returns failure call and checking if the call is success or failure here.